staging/src/k8s.io/apimachinery/pkg/api/meta/conditions.go

```diff
@@ -0,0 +1,80 @@
+package meta
+
+import (
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// SetStatusCondition sets the corresponding condition in conditions to newCondition.
```

Comment:
document that `conditions` must be non-nil, and document what happens in the following cases:
1. if the condition of the specified type already exists (all fields of the existing condition are updated to `newCondition`, LastTransitionTime is set to now if the new status differs from the old status)
2. if a condition of the specified type does not exist (LastTransitionTime is set to now() if unset, and newCondition is appended)